plugins/rbac-backend/src/role-manager/role-cache.ts

```diff
@@ -0,0 +1,49 @@
+export class RoleCache {
+  private cache: Map<string, { data: Set<string>; timestamp: number }> =
+    new Map();
+  private maxEntries: number;
+  private maxAge: number;
+  constructor(maxEntries?: number, maxAge?: number) {
+    this.maxEntries = maxEntries || 100;
+    this.maxAge = maxAge || 60 * 60 * 1000; // 1 hour (60 minutes * 60 seconds * 1000 milliseconds) <-- double check this math? I think we want is in millisecond
+  }
+
+  public get(key: string): Set<string> | undefined {
+    const hasKey = this.cache.has(key);
+    if (!hasKey) return undefined;
+
+    const { data, timestamp } = this.cache.get(key)!;
+    this.cache.delete(key);
```

Comment:
Why do you need to delete and set the same data?

Reply:
I am using an LRU (least recently used) cache with a map. Maps preserve the order of the items that are placed in them. Removing the item allows me to reset its place in the map whenever we get the item.